scipy/stats/_qmc.py

```diff
@@ -898,6 +988,151 @@ def random(self, n: IntNumber = 1) -> np.ndarray:
         return samples  # type: ignore[return-value]
 
 
+class OptimalLatinHypercube(QMCEngine):
+    """Optimal Latin hypercube sampling (OLHS).
+
+    Optimize the sample by doing random permutations of coordinates to lower
+    the centered discrepancy [1]_. The resulting sample is still a LHS.
+
+    Parameters
+    ----------
+    d : int
+        Dimension of the parameter space.
+    start_sample : array_like (n, d), optional
+        Initial sample to optimize. `LatinHypercube`
+        is used to generate a first design otherwise.
+    n_perturbations : int, optional
+        Number of perturbation to perform. Default is 10000.
+    seed : {None, int, `numpy.random.Generator`}, optional
+        If `seed` is None the `numpy.random.Generator` singleton is used.
+        If `seed` is an int, a new ``Generator`` instance is used,
+        seeded with `seed`.
+        If `seed` is already a ``Generator`` instance then that instance is
+        used.
+
+    Notes
+    -----
+    `n_perturbations` set of random permutations are performed aginst a LHS
+    sample. The best design based on the centered discrepancy is constantly
+    update. Centered discrepancy-based design shows better space filling
+    robustness toward 2D and 3D subprojections. Distance-based design shows
+    better space filling but less robustness to subprojections [2]_.
```

Comment:
```
A sequence of random perturbations are performed on an LHS sample.
Each perturbation consists of swapping a random coordinate of a randomly-chosen
sample point with the corresponding coordinate of another randomly-chosen sample
point, preserving the Latin Hypercube properties of the sample.
Perturbations that reduce the centered discrepancy of the sample are preserved;
perturbations that increase the centered discrepancy are rejected. After
`n_perturbations` are considered, the process terminates, resulting in a new LHS
with substantially lower centered discrepancy.
```
I'm not sure if the notes about centered discrepancy are needed here. It seems like it's being compared to "distance-based design", but there is no option to optimize a distance-based design, so why compare the two here? 
> Centered discrepancy-based design shows better space..

Better compared to what?

> Distance-based design

Is this an alternative to centered discrepancy?

Reply:
My intent was to say that the centered discrepancy was a good choice and on purpose. You could chose to optimize the wrap around discrepancy, but from the linked article they have seen worse performance. To have the nice update function, you also need to derive it. It has been done for not all measures (and I just coded and fixed this one).

Here we don't have distance-based measures (I have an open issue for that but very easy so for now I leave it for Sprints...). Distance measures are simpler things like L-inf or mean distance between points, etc.

Reply:
I simplified the doc here.